src/storage/store/shard.rs

```diff
@@ -118,25 +124,70 @@ pub fn put_shard_chunk(db: &RocksDB, shard_chunk: ShardChunk) -> Result<(), Shar
     Ok(())
 }
 
-#[derive(Default)]
+pub fn get_shard_chunks_in_range(
+    db: &RocksDB,
+    page_options: &PageOptions,
+    start_block_number: u64,
+    stop_block_number: Option<u64>,
+) -> Result<ShardPage, ShardStorageError> {
+    let start_primary_key = make_shard_key(start_block_number);
+    let stop_prefix = stop_block_number.map(|block_number| make_shard_key(block_number));
+
+    get_shard_page_by_prefix(db, page_options, Some(start_primary_key), stop_prefix)
+}
+
+#[derive(Default, Clone)]
 pub struct ShardStore {
-    db: RocksDB,
+    db: Arc<RocksDB>,
 }
 
 impl ShardStore {
-    pub fn new(db: RocksDB) -> ShardStore {
+    pub fn new(db: Arc<RocksDB>) -> ShardStore {
         ShardStore { db }
     }
 
     pub fn put_shard_chunk(&self, shard_chunk: ShardChunk) -> Result<(), ShardStorageError> {
         put_shard_chunk(&self.db, shard_chunk)
     }
 
+    pub fn get_last_shard_chunk(&self) -> Result<Option<ShardChunk>, ShardStorageError> {
+        get_last_shard_chunk(&self.db)
+    }
+
     pub fn max_block_number(&self) -> Result<u64, ShardStorageError> {
         let current_height = get_current_height(&self.db)?;
         match current_height {
             None => Ok(0),
             Some(height) => Ok(height),
         }
     }
+
+    pub fn get_shard_chunks(
```

Comment:
maybe it's time for some unit tests on things like this? Then again maybe it's too early. Perhaps some TODOs where we'd like to have unit tests (assuming these functions survive further refactoring).

Reply:
Yeah agreed-- I think it's worth unit testing. I'll write tests after we discuss how to handle the sync edge cases today? That might change this interface somewhat.